File: programming/nonmainstream/rexx/Java/RexxParserBase.java
import org.antlr.v4.runtime.*;

public abstract class RexxParserBase extends Parser
{
   public RexxParserBase(TokenStream input)
   {
      super(input);
   }

   public final boolean is_prev_token_whitespace() {
      int tokenIndex = getCurrentToken().getTokenIndex();
      int prevTokenType = getTokenStream().get(tokenIndex-1).getType();
      return (prevTokenType == RexxLexer.WHITESPACES);
   }

   public final boolean is_prev_token_not_whitespace() {
      return !is_prev_token_whitespace();
   }

}

File: rexx/Python3/RexxParserBase.py
from antlr4 import *
import RexxLexer

class RexxParserBase(Parser):
   def is_prev_token_whitespace(self):
      tokenIndex = self.getCurrentToken().tokenIndex
      prevTokenType = self.getTokenStream().get(tokenIndex-1).type
      return (prevTokenType is RexxLexer.RexxLexer.WHITESPACES)

   def is_prev_token_not_whitespace(self):
      return not self.is_prev_token_whitespace()
